app/models/post.rb

```diff
@@ -1,3 +1,15 @@
 class Post < ApplicationRecord
-  belongs_to :user
+  belongs_to :user, class_name: 'User', foreign_key: 'user_id'
+  has_many :comments
+  has_many :likes
+
+  after_save :update_post_counter
+
+  def update_post_counter
+    user.increment_counter(:post_counter, 1)
+  end
```

Comment:
- [ ] I think it would be better if your use the ``increment!`` method to update the comments counter as the current one doesn't work on my side. you can update it to something like: ``user.increment!(:post_counter)``